pkg/bloomgateway/bloomgateway_test.go

```diff
@@ -263,9 +264,13 @@ func TestBloomGateway_FilterChunkRefs(t *testing.T) {
 		// replace store implementation and re-initialize workers and sub-services
 		bqs, data := createBlockQueriers(t, 5, now.Add(-8*time.Hour), now, 0, 1024)
 		gw.bloomStore = newMockBloomStore(bqs)
+		// randomize between sequentical processing and processing using callback
+		gw.workerConfig.processBlocksSequentially = rand.Float32() > 0.5
```

Comment:
Would it be better to have two tests, one that did sequential, and one that used the callback?  I'm only thinking of flaky test cases, and that if there was something strange happening here, it may be hard to pin down.